tools/mage/dev.go

```diff
@@ -70,10 +78,97 @@ func execDockerCompose(args ...string) error {
 	return err
 }
 
+func execDockerComposeWithOutput(filepath string, args ...string) error {
+	output, err := sh.Output("docker-compose", dockerComposeFlags(args...)...)
+	if err != nil {
+		return err
+	}
+	message := []byte(output)
+	err = ioutil.WriteFile(filepath, message, 0644)
+	return err
+}
+
+// SQLStart starts the SQL database of the development environment.
+func (Dev) SQLStart() error {
+	if mg.Verbose() {
+		fmt.Println("Starting SQL database")
+	}
+	if err := execDockerCompose(append([]string{"up", "-d"}, sqlDatabase)...); err != nil {
+		return err
+	}
+	return execDockerCompose("ps")
+}
+
+// SQLStop stops the SQL database of the development environment.
+func (Dev) SQLStop() error {
+	if mg.Verbose() {
+		fmt.Println("Stopping SQL database")
+	}
+	return execDockerCompose(append([]string{"stop"}, sqlDatabase)...)
+}
+
+// SQLRestoreSnapshot restores the previously taken snapshot, thus restoring all previously
+// snapshoted databases.
+func (d Dev) SQLRestoreSnapshot() error {
+	mg.Deps(Dev.SQLStop)
+	if mg.Verbose() {
+		fmt.Println("Restoring DB snapshot")
+	}
+	to := filepath.Join(devDataDir, "cockroach")
+	from := filepath.Join(devDataDir, "cockroach-snap")
+	if err := os.RemoveAll(to); err != nil {
+		return err
+	}
+	if err := sh.Copy(from, to); err != nil {
+		return err
+	}
+	return d.SQLStart()
+}
+
+// DBDump performs an SQL database dump of the dev database to the .cache folder.
+func (Dev) DBDump() error {
+	if mg.Verbose() {
+		fmt.Println("Saving database dump")
+	}
+	if err := os.MkdirAll(".cache", 0755); err != nil {
+		return err
+	}
+	return execDockerComposeWithOutput(filepath.Join(".cache", "sqldump.sql"), "exec", "-T", "cockroach", "./cockroach", "dump", devDatabaseName, "--insecure")
```

Comment:
Moreover, this is the only usage of execDockerComposeWithOutput in the whole codebase - just remove the helper and use `execDockerCompose` explicitly here